### ISSUE TITLE
How is resource_limit_elderly_or_disabled_income_twice_fpl intended to be used?

### ISSUE THREAD

**Issue by benmolin** [18F/snap-js-api-prototype#35]
In asset_test.js ```resource_limit_elderly_or_disabled``` and ```resource_limit_non_elderly_or_disabled``` are supported. ```resource_limit_elderly_or_disabled_income_twice_fpl``` appears to currently not be factored into the api.

The [BBCE](https://fns-prod.azureedge.net/sites/default/files/resource-files/BBCEStatesChart%28May2020%29.pdf) chart says for this:

> In these States, households with seniors or people with disabilities and gross income under 200 percent of poverty
do not face an asset limit. Those over 200 percent of poverty are not categorically eligible and do face a $3,500 asset
limit.

It seems like most of the time it's a non-factor, because if you're over 200% FPL you'd probably be rejected by the gross income test before you even got to the asset test. So unless a certain state does have a >200% FPL gross income test, this doesn't come into consideration. It doesn't appear that any do from the PDF, but some states might have a higher income limit than listed for elderly or disabled.

If PA had a 250% gross income limit for elderly or disabled, I am guessing you would need to change [asset_test.js](https://github.com/18F/snap-js-api-prototype/blob/main/src/tests/asset_test.js#L28) to add in a rule:

```
 if (gross_income > 200% FLP) & (senior_or_disabled_in_household): 
    apply_asset_test()
else:
   ...
```

**Comment by alexsoble** [comment]
@benmolin You're right that this isn't factored into the API yet — since we aren't working with any partners in Illinois at this point, fleshing out every IL scenario hasn't been the highest priority. Virginia doesn't have a special resource limit for households with an elderly or disabled member whose income is twice FPL. 

Since income and assets are separate concepts, I wouldn't add any income checks in the `asset_test.js` file.

**Comment by benmolin** [comment]
@alexsoble I believe I better understand the concept since I raised this issue. For [BBCE States with Footnote 2](https://fns-prod.azureedge.net/sites/default/files/resource-files/BBCEStatesChart%28May2020%29.pdf) there is some crossover between income and assets tests -- meaning that your status one one will effect the other. 

More specifically, in some BBCE states the only circumstance that there is an asset test is if the household has an elderly or disabled member and the household is above 200% of the federal poverty limit. First, that means we need to pass gross income (along with other variables) into ``asset_test.js`` file, as without it we will not be able to figure out whether the household automatically passes the AssetTest, or whether they have to pass a $3500 limit.

Here is my proposed code, to be placed in ``asset_test.js``:

```

// If HHEOD, and Gross Income is 200% FPL, and it applies in this state, there is an asset test
if (this.household_includes_elderly_or_disabled &&
    this.uses_bbce &&
    this.has_resource_limit_elderly_or_disabled_income_twice_fpl &&
    (this.gross_income > this.gross_monthly_income_limit_twice_fpl)) {

    // Asset test using $3,500
    const below_resource_limit = this.resources <= 3500;
    let explanation = `Since this household has resources of $${this.resources}, it <strong>${below_resource_limit ? 'meets' : 'does not meet'}</strong> the asset test.`

    return {
        'name': 'Asset Test',
        'result': below_resource_limit,
        'explanation': [
            `Since the household includes an elderly or disabled member and has gross income above 200% of the federal poverty guidelines, ${this.state_or_territory} requires the household pass a $3500 asset limit. ` + explanation
        ],
        'sort_order': 4,
        'type': 'test'
    };

};
```
